### ISSUE TITLE
[REQ][TYPESCRIPT-AXIOS] Add option to not stringify JSON data in axios-typescript generator

### ISSUE THREAD

**Issue by aranoe** [OpenAPITools/openapi-generator#5717]
## Problem Statement

The generator always serializes objects to string with `JSON.stringify`. This is not always wanted, which is why it should configurable. In my case I wanted to write an `AxiosTransformer`, which transforms JavaScript `Date` types to the correct format. In order to do this now, I have to `JSON.parse` the data and then check it. 

Here is the line in question:
https://github.com/OpenAPITools/openapi-generator/blob/ce49fe587fe6ab4dcc2275590f49fb0b09777710/modules/openapi-generator/src/main/resources/typescript-axios/apiInner.mustache#L199

## Prefered Solution
Add new option to the config, when generating the client:
```
{
  "stringifyRequestData": false
}
```

**Comment by amakhrov** [comment]
Note that typescript-axios currently doesn't support working with JS Date objects. date/datetime fields in the spec are represented as `string` in the generated typescript interfaces

**Comment by aranoe** [comment]
Thats not true. You can use type mappings to convert `date` types to `Date` via following option:

`--type-mappings date=Date`

That way the date properties in the interfaces can be `Date` instead of `string`.

**Comment by rvowles** [comment]
I just ran smack into  this exact this problem. I would prefer a mechanism where we had the option to in-situ json encode and decode properly and also do deep copies.

**Comment by Donorlin** [comment]
Any news about this? It would be a really nice feature to be able to inject either own JSON.stringify's replacer (second argument of JSON.stringify) or be able to turn off serialization completely. 

In most Axios cases serialization is done via Axios's request transformers and because of this issue, all objects are serialized in generated API code, not in request transformers. (request transformers are working with already serialized objects -> strings and doing nothing).

**Comment by nadilas** [comment]
I just ran into this. It would be a nice option to have.

**Comment by rvowles** [comment]
I wrote my own generator - you just add it to your classpath. The wheels turn too slowly here. 

https://github.com/rvowles/openapi3-typescript-axios

https://search.maven.org/artifact/com.bluetrainsoftware.maven/openapi-tsaxios-generator/1.1/jar - should be able to download it
